### PR TITLE
Update when did it happen form

### LINKED ISSUE #2159: Update When Did It Happen Form
Update WhenDidItHappenForm.js to use Formik, Yup, and Bootstrap as documented in issue #2136. These changes have been made to HowDidItStartForm.js, you can them here - #2138.  

If radio component has not already been implemented it will required - documentation [here](https://react-bootstrap.github.io/components/forms/#forms-form-check).

This form uses date validation, this should be refactored to use Yup - [Yup Date](https://github.com/jquense/yup#date) and [Yup Extend Type](https://github.com/jquense/yup#extending-schema-types).

### PR DESCRIPTION
**Pull request by justinr86**
Fixes #2159 

# Description

When Did It Start form has been refactored to use React Bootstrap, Formik, and Yup. This update also modifies other components to resolve minor UI issues.

Validation has been improved for dates and an ErrorSummary component has been created. 

Also, a FormikListener component has been created to manage form level status.

# Any new packages installed?

moment.js

# Required followup work

Discussion regarding validation and error messages.

# Checklist:

- [ ] I have updated the azurescript.sh with any **new environment variables** and added them to the appsettings
- [ ] I have looked at my code on GitHub and it all looks good (ex: no random commented out code or console.logs)
- [ ] I have added and needed tests for my changes (in particular for new screens)
- [ ] I have added a comment to any confusing code
- [ ] I have added documentation to any modified front-end code. (Or added missing documentation)

### CODE REVIEW

**Comment by ying-chen-cds** [comment]
Did you use error summary in when did it happen page?

**Comment by ying-chen-cds** [comment]
> Did you use error summary in when did it happen page?

OK, just checked the code, seems not used.

**Comment by justinr86** [comment]
> > Did you use error summary in when did it happen page?
> 
> OK, just checked the code, seems not used.

I'm not sure what you mean. There is an ErrorSummary component used, it is exported from f2\src\components\formik\alert\index.js.

**Comment by ying-chen-cds** [comment]
> > > Did you use error summary in when did it happen page?
> > 
> > 
> > OK, just checked the code, seems not used.
> 
> I'm not sure what you mean. There is an ErrorSummary component used, it is exported from f2\src\components\formik\alert\index.js.

I tested 

> > > Did you use error summary in when did it happen page?
> > 
> > 
> > OK, just checked the code, seems not used.
> 
> I'm not sure what you mean. There is an ErrorSummary component used, it is exported from f2\src\components\formik\alert\index.js.

I checked out your branch, and I didn't see you use ErrorSummary component in whenDidItHappenForm. I just want to see how it looks like in the UI.

**Comment by ying-chen-cds** [comment]
Suggest to use red color for "Please fix the errors on this page." above the submit button.
image

**Comment by justinr86** [comment]
> > > > Did you use error summary in when did it happen page?
> > > 
> > > 
> > > OK, just checked the code, seems not used.
> > 
> > 
> > I'm not sure what you mean. There is an ErrorSummary component used, it is exported from f2\src\components\formik\alert\index.js.
> 
> I tested
> 
> > > > Did you use error summary in when did it happen page?
> > > 
> > > 
> > > OK, just checked the code, seems not used.
> > 
> > 
> > I'm not sure what you mean. There is an ErrorSummary component used, it is exported from f2\src\components\formik\alert\index.js.
> 
> I checked out your branch, and I didn't see you use ErrorSummary component in whenDidItHappenForm. I just want to see how it looks like in the UI.

The ErrorSummary component is not used for field level errors, these errors appear as the user is typing. The ErrorSummary component is used for form level errors. To display this enter an invalid date that will pass field level validation, such as {Day: 30, Month: 02, Year: 2020}. This will fail because there is no 30th day of February.

**Comment by ying-chen-cds** [comment]
> > > > > Did you use error summary in when did it happen page?
> > > > 
> > > > 
> > > > OK, just checked the code, seems not used.
> > > 
> > > 
> > > I'm not sure what you mean. There is an ErrorSummary component used, it is exported from f2\src\components\formik\alert\index.js.
> > 
> > 
> > I tested
> > > > > Did you use error summary in when did it happen page?
> > > > 
> > > > 
> > > > OK, just checked the code, seems not used.
> > > 
> > > 
> > > I'm not sure what you mean. There is an ErrorSummary component used, it is exported from f2\src\components\formik\alert\index.js.
> > 
> > 
> > I checked out your branch, and I didn't see you use ErrorSummary component in whenDidItHappenForm. I just want to see how it looks like in the UI.
> 
> The ErrorSummary component is not used for field level errors, these errors appear as the user is typing. The ErrorSummary component is used for form level errors. To display this enter an invalid date that will pass field level validation, such as {Day: 30, Month: 02, Year: 2020}. This will fail because there is no 30th day of February.

I see. Thanks!

File: f2/src/components/formik/formikListener/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  if (formError) {
+    if (status.errorValues) {
+      fieldsUpdated =
+        JSON.stringify(values) !== JSON.stringify(status.errorValues)
+    } else {
+      setStatus({ errors: formError, errorValues: values })
+    }
+
+    if (fieldsUpdated) {
```
Comment (sdzhangtao) [inline f2/src/components/formik/formikListener/index.js:16]: Is this variable necessary?

Comment (sdzhangtao) [reply f2/src/components/formik/formikListener/index.js:16]: import { useFormikContext } from 'formik'

export const FormikListener = () => {
  const { values, status, setStatus } = useFormikContext()
  const formError = status.errors
  let fieldsUpdated = false

  if (formError) {
    if (status.errorValues) {
      if(JSON.stringify(values) !== JSON.stringify(status.errorValues))
        setStatus({ errors: '' })
    } else {
      setStatus({ errors: formError, errorValues: values })
    }

  }
  return null
}

**Pull request merged**